src/helpers/check_diff.jl

```diff
@@ -0,0 +1,136 @@
+
+# Fits \hat{y} = a + b x.
+# Returns a, b, and the R^2 of the fit.
+function solve_simple_lm(x, y)
+    # See https://en.wikipedia.org/wiki/Simple_linear_regression
+    correlation = cor(x, y)
+    beta = std(y) / std(x) * correlation
+    alpha = mean(y) - beta * mean(x)
+    r2 = size(x, 1) <= 2 ? 1 : correlation^2
+    return (alpha, beta, r2)
+end
+
+function linearize_function(x, y; r2_cutoff=0.99)
```

Comment:
I am not sure what this function is for - it is undocumented and has very few comments – also I would prefer having this without the `DataFrame` dependency, what is it used fo? (If the function would be more documented I could propose an alternative...)

Reply:
Ah now I see. We should try to do that without a Dataframe.

My idea (but I will only be able to program something like that maybe next weekend) would be to look for the longest interval (stored in log scale) that fulfils the slope (up to the precision/deviatin I tried to set up over on the other PR) and return that.

If that is the whole range the user reports - all is fine and we return true, if not we give the user the hint (as. `@warn`. maybe?) what the longest interval is and return false (or throw an error depending on what the user specified), so that an user (without the plot) still gets some idea of how good his differential is.

What do you think?